pkg/controllers/nodeclass/status/ami.go

```diff
@@ -44,6 +48,16 @@ func (a *AMI) Reconcile(ctx context.Context, nodeClass *v1.EC2NodeClass) (reconc
 		nodeClass.StatusConditions().SetFalse(v1.ConditionTypeAMIsReady, "AMINotFound", "AMISelector did not match any AMIs")
 		return reconcile.Result{}, nil
 	}
+	// If an alias is configured for automatic upgrades, and we receive a deprecated AMI, we should invalidate the SSM
+	// cache ahead of the 24 hour expiration period. This ensures Karpenter can react quickly to AMI deprecations due
+	// to regressions in EKS-optimized AMIs.
+	if alias := nodeClass.Alias(); alias != nil && alias.Version == v1.AliasVersionLatest && lo.ContainsBy(amis, func(a amifamily.AMI) bool { return a.Deprecated }) {
+		k8sVersion, err := a.versionProvider.Get(ctx)
```

Comment:
I didn't want the SSM provider to need to reason about how to construct SSM parameters since the k8s version is in different parts of different SSM paths. That responsibility is delegated to the ami families. None of this should be required with the separate invalidation controller approach though.